lib/cartopy/tests/mpl/test_boundary.py

```diff
@@ -0,0 +1,94 @@
+# Copyright Crown and Cartopy Contributors
+#
+# This file is part of Cartopy and is released under the BSD 3-clause license.
+# See LICENSE in the root of the repository for full licensing details.
+from itertools import chain
+
+from matplotlib.path import Path
+import matplotlib.pyplot as plt
+import numpy as np
+import pytest
+
+import cartopy.crs as ccrs
+
+
+circle_verts = np.array([
+    (21.33625905034713, 41.90051020408163),
+    (20.260167503134653, 36.31721708143521),
+    (17.186058185078757, 31.533809612693403),
+    (12.554340705353228, 28.235578526280886),
+    (7.02857405747471, 26.895042042418392),
+    (1.4004024147599825, 27.704250959518802),
+    (-3.5238590865749853, 30.547274662874656),
+    (-7.038740387110195, 35.0168098237746),
+    (-8.701144846177232, 41.42387784534415),
+    (-7.802741418346137, 47.03850217758886),
+    (-4.224119444187849, 52.606946662776494),
+    (0.5099122186645388, 55.75656240544797),
+    (6.075429329647158, 56.92110282927732),
+    (11.675092899771812, 55.933731058974054),
+    (16.50667197715324, 52.93590205611499),
+    (19.87797456957319, 48.357097196578444),
+    (21.33625905034713, 41.90051020408163)
```

Comment:
Should we make one of these non-closed initially so you show that they are closed by the `to_polygons` call?

Reply:
Yes we definitely should! 

Especially because I just realized in the process that the use of `Path.clip_to_bbox()` actually explicitly requires the use of  closed polygons 😅 

> Clip the path to the given bounding box.
>
> The path must be made up of one or more closed polygons. This algorithm will not behave correctly for unclosed paths.

So `_ensure_closed_polyons` actually needs to be applied also on `ax.set_boundary`... e.g.:  
```self._original_path = self._ensure_path_closed(path)`

Reply:
Actually it should be enough to ensure the initial path is properly closed and then let `clip_to_bbox()` handle the rest...
This way we completely avoid a potential hot-loop when zooming/panning since `_ensure_path_closed` is only called once for the initial path passed to `ax.set_boundary`